src/sysdir.c

```diff
@@ -189,7 +189,7 @@ int git_sysdir_global_init(void)
 	for (i = 0; !error && i < ARRAY_SIZE(git_sysdir__dirs); i++)
 		error = git_sysdir__dirs[i].guess(&git_sysdir__dirs[i].buf);
 
-	git__on_shutdown(git_sysdir_global_shutdown);
+	git_runtime_shutdown_register(git_sysdir_global_shutdown);
```

Comment:
We're ignoring the function's return code here.

Reply:
Fixed.